behaviour/graql/reasoner/recursion.feature

```diff
@@ -0,0 +1,2025 @@
+      (Q-from: $a0, Q-to: $b1_9) isa Q;
+      (Q-from: $a0, Q-to: $b1_10) isa Q;
+
+
+      # (Q-from: $b{i}_{j}, Q-to: $b{i+1}_{j}) isa Q; for 1 <= j <= n; for 1 <= i <= m
+      (Q-from: $b1_1, Q-to: $b2_1) isa Q;
+      (Q-from: $b2_1, Q-to: $b3_1) isa Q;
+      (Q-from: $b3_1, Q-to: $b4_1) isa Q;
+      (Q-from: $b4_1, Q-to: $b5_1) isa Q;
+      (Q-from: $b5_1, Q-to: $b6_1) isa Q;
+
+      (Q-from: $b1_2, Q-to: $b2_2) isa Q;
+      (Q-from: $b2_2, Q-to: $b3_2) isa Q;
+      (Q-from: $b3_2, Q-to: $b4_2) isa Q;
+      (Q-from: $b4_2, Q-to: $b5_2) isa Q;
+      (Q-from: $b5_2, Q-to: $b6_2) isa Q;
+
+      (Q-from: $b1_3, Q-to: $b2_3) isa Q;
+      (Q-from: $b2_3, Q-to: $b3_3) isa Q;
+      (Q-from: $b3_3, Q-to: $b4_3) isa Q;
+      (Q-from: $b4_3, Q-to: $b5_3) isa Q;
+      (Q-from: $b5_3, Q-to: $b6_3) isa Q;
+
+      (Q-from: $b1_4, Q-to: $b2_4) isa Q;
+      (Q-from: $b2_4, Q-to: $b3_4) isa Q;
+      (Q-from: $b3_4, Q-to: $b4_4) isa Q;
+      (Q-from: $b4_4, Q-to: $b5_4) isa Q;
+      (Q-from: $b5_4, Q-to: $b6_4) isa Q;
+
+      (Q-from: $b1_5, Q-to: $b2_5) isa Q;
+      (Q-from: $b2_5, Q-to: $b3_5) isa Q;
+      (Q-from: $b3_5, Q-to: $b4_5) isa Q;
+      (Q-from: $b4_5, Q-to: $b5_5) isa Q;
+      (Q-from: $b5_5, Q-to: $b6_5) isa Q;
+
+      (Q-from: $b1_6, Q-to: $b2_6) isa Q;
+      (Q-from: $b2_6, Q-to: $b3_6) isa Q;
+      (Q-from: $b3_6, Q-to: $b4_6) isa Q;
+      (Q-from: $b4_6, Q-to: $b5_6) isa Q;
+      (Q-from: $b5_6, Q-to: $b6_6) isa Q;
+
+      (Q-from: $b1_7, Q-to: $b2_7) isa Q;
+      (Q-from: $b2_7, Q-to: $b3_7) isa Q;
+      (Q-from: $b3_7, Q-to: $b4_7) isa Q;
+      (Q-from: $b4_7, Q-to: $b5_7) isa Q;
+      (Q-from: $b5_7, Q-to: $b6_7) isa Q;
+
+      (Q-from: $b1_8, Q-to: $b2_8) isa Q;
+      (Q-from: $b2_8, Q-to: $b3_8) isa Q;
+      (Q-from: $b3_8, Q-to: $b4_8) isa Q;
+      (Q-from: $b4_8, Q-to: $b5_8) isa Q;
+      (Q-from: $b5_8, Q-to: $b6_8) isa Q;
+
+      (Q-from: $b1_9, Q-to: $b2_9) isa Q;
+      (Q-from: $b2_9, Q-to: $b3_9) isa Q;
+      (Q-from: $b3_9, Q-to: $b4_9) isa Q;
+      (Q-from: $b4_9, Q-to: $b5_9) isa Q;
+      (Q-from: $b5_9, Q-to: $b6_9) isa Q;
+
+      (Q-from: $b1_10, Q-to: $b2_10) isa Q;
+      (Q-from: $b2_10, Q-to: $b3_10) isa Q;
+      (Q-from: $b3_10, Q-to: $b4_10) isa Q;
+      (Q-from: $b4_10, Q-to: $b5_10) isa Q;
+      (Q-from: $b5_10, Q-to: $b6_10) isa Q;
+      """
+#    When materialised keyspace is completed
+    Then for graql query
+      """
+      match
+        (P-from: $x, P-to: $y) isa P;
+        $x has index 'a0';
+      get $y;
+      """
+#    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 60
+    Then answer set is equivalent for graql query
+      """
+      match $y isa b-entity; get;
+      """
+#    Then materialised and reasoned keyspaces are the same size
+
+
+  # TODO: re-enable all steps when resolvable (currently takes too long) (#75)
+  Scenario: Nguyen's test
+
+    test3 from Nguyen (similar to test 6.5 from Cao):
```

Comment:
Madalinska-Bugaj, Nguyen - A Generalized QSQR Evaluation Method for Horn Knowledge Bases

Reply:
Citation added